build/Build.cs

```diff
@@ -0,0 +1,142 @@
+using Nuke.Common.Utilities.Collections;
+using static Nuke.Common.IO.FileSystemTasks;
+using static Nuke.Common.Tools.DotMemoryUnit.DotMemoryUnitTasks;
+using static Nuke.Common.Tools.DotNet.DotNetTasks;
+using static Nuke.Common.Tools.GitLink.GitLinkTasks;
+
+[CheckBuildProjectConfigurations]
+[ShutdownDotNetAfterServerBuild]
+class Build : NukeBuild
+{
+    /// Support plugins are available for:
+    ///   - JetBrains ReSharper        https://nuke.build/resharper
+    ///   - JetBrains Rider            https://nuke.build/rider
+    ///   - Microsoft VisualStudio     https://nuke.build/visualstudio
+    ///   - Microsoft VSCode           https://nuke.build/vscode
+
+    [Parameter("Configuration to build - Default is 'Debug' (local) or 'Release' (server)")]
+    readonly Configuration Configuration = IsLocalBuild ? Configuration.Debug : Configuration.Release;
+
+    [Solution(GenerateProjects = true)] readonly Solution Solution;
+
+    [Parameter("Whether to auto-detect the branch name - this is okay for a local build, but should not be used under CI.")]
+    readonly bool AutoDetectBranch = IsLocalBuild;
+
+    [Parameter("Branch name for OctoVersion to use to calculate the version number. Can be set via the environment variable OCTOVERSION_CurrentBranch.",
+        Name = "OCTOVERSION_CurrentBranch")]
+    readonly string BranchName;
+
+    [OctoVersion(UpdateBuildNumber = true, BranchParameter = nameof(BranchName),
+        AutoDetectBranchParameter = nameof(AutoDetectBranch), Framework = "net6.0")]
+    readonly OctoVersionInfo OctoVersionInfo;
+
+    static AbsolutePath LocalPackagesDir => RootDirectory / ".." / "LocalPackages";
+
+    static AbsolutePath SourceDirectory => RootDirectory / "source";
+
+    static AbsolutePath ArtifactsDirectory => RootDirectory / "artifacts";
+
+    Target Clean => _ => _
+        .Executes(() =>
+        {
+            SourceDirectory.GlobDirectories("**/bin", "**/obj").ForEach(DeleteDirectory);
+            EnsureCleanDirectory(ArtifactsDirectory);
+        });
+
+    Target Restore => _ => _
+        .DependsOn(Clean)
+        .Executes(() =>
+        {
+            DotNetRestore(s => s
+                .CombineWith(ss => ss
+                    .SetProjectFile(Solution.Halibut))
+                .CombineWith(ss => ss
+                    .SetProjectFile(Solution.Halibut_Tests)));
+        });
+
+    Target Compile => _ => _
+        .DependsOn(Restore)
+        .Executes(() =>
+        {
+            DotNetBuild(s => s
+                .SetConfiguration(Configuration)
+                .SetVersion(OctoVersionInfo.FullSemVer)
+                .SetInformationalVersion(OctoVersionInfo.InformationalVersion)
+                .EnableNoRestore()
+                .CombineWith(ss => ss
+                    .SetProjectFile(Solution.Halibut))
+                .CombineWith(ss => ss
+                    .SetProjectFile(Solution.Halibut_Tests)));
+        });
+
+    [PublicAPI]
+    Target TestWindows => _ => _
```

Comment:
I've separated windows/linux so that we can still run the dotmemory tests on windows, however dotmemory isn't supported on linux so we just run a `dotnet test` instead.